x-pack/plugins/cloud_security_posture/server/routes/configuration/update_rules_configuration.ts

```diff
@@ -65,21 +65,26 @@ export const getCspRules = (
   });
 };
 
+const getEnabledRulesByBenchmark = (rules: SavedObjectsFindResponse<CspRule>['saved_objects']) =>
+  rules.reduce<CspRulesConfiguration['data_yaml']['activated_rules']>(
+    (benchmarks, rule) => {
+      const benchmark = benchmarks[rule.attributes.metadata.benchmark.id];
+      if (!rule.attributes.enabled || !benchmark) return benchmarks;
+
+      benchmark.push(rule.attributes.metadata.rego_rule_id);
+
+      return benchmarks;
+    },
+    { cis_k8s: [], cis_eks: [] }
+  );
```

Comment:
and this is how we generate the activated rules.

the first iteration of this code used to grab the first item and check for the `id` and as it's (currently) the same for all rules, it used it as the key and if it doesn't exists it used to throw. 
although it works, @kfirpeled suggested we could make this function more robust by simply building the array we need by iterating, which enables us to pass 2 types (even though currently it's just 1) and without the need to account for non-existing benchmark id. much better without the throwing. 

thanks @kfirpeled ! 💯